sdk/core/corehttp/corehttp/runtime/pipeline/_base.py

```diff
@@ -123,7 +129,9 @@ def __init__(
         self._transport = transport
 
         for policy in policies or []:
-            if isinstance(policy, SansIOHTTPPolicy):
+            if isinstance(policy, HTTPPolicy):
```

Comment:
what would the error/behavior be today if I send in a policy that falls in to the last elif?

Reply:
updated!

Reply:
If it were to go into the last elif, it would raise an AttributeError that the credential has no attribute 'send'.

Reply:
reverting back - the last elif accounted for the case where the `send` is a callable attribute in the policy, but isn't necessarily a subtype of the protocol. The callable send attribute should still work. For example, this Mock policy in the auth test: https://github.com/Azure/azure-sdk-for-python/blob/b5c584566343a496bbf02ca9e8c71af922547959/sdk/core/corehttp/tests/test_authentication.py#L31

Reply:
actually, since the policies parameter specifies that the type has to be Optional[Union[HTTPPolicy, SansIOHTTPPolicy]], if this Mock policy doesn't follow the protocol, it _should_ fail I think? So should the test be updated so that the Mock policy passed in follows the protocol?

Reply:
removing isinstance + last elif check b/c of redundancy

Reply:
The Mock should follow the policy protocol, and should fail